src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -133,8 +135,12 @@ public boolean equals(Object other) {
         }
 
         CommandResult otherCommandResult = (CommandResult) other;
+
+        boolean isSummary = !displaySummary || !otherCommandResult.displaySummary;
+
         return feedbackToUser.equals(otherCommandResult.feedbackToUser)
                 && showCommandSummary == otherCommandResult.showCommandSummary
+                && (isSummary || summary.equals(otherCommandResult.summary))
```

Comment:
In this case, when comparing CommandResult with displaySummary = false and otherCommandResult with displaySummary = true (or vice versa), isSummary will return true and the summary equality check will be bypassed and line 143 will return true (even if summary.equals(otherCommandResult.summary) returns false.

Reply:
Yup! I have updated it such that both the display summary and summary itself need to be compared to be equal